dao/src/main/java/org/thingsboard/server/dao/edge/BaseEdgeEventService.java

```diff
@@ -19,24 +19,33 @@
 import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
+import org.thingsboard.server.common.data.EntityType;
 import org.thingsboard.server.common.data.edge.EdgeEvent;
 import org.thingsboard.server.common.data.id.EdgeId;
 import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.common.data.limit.LimitedApi;
 import org.thingsboard.server.common.data.page.PageData;
 import org.thingsboard.server.common.data.page.TimePageLink;
+import org.thingsboard.server.common.msg.tools.TbRateLimitsException;
 import org.thingsboard.server.dao.service.DataValidator;
+import org.thingsboard.server.dao.util.limits.RateLimitService;
 
 @Service
 @Slf4j
 @AllArgsConstructor
 public class BaseEdgeEventService implements EdgeEventService {
 
     private final EdgeEventDao edgeEventDao;
-
+    private final RateLimitService rateLimitService;
     private final DataValidator<EdgeEvent> edgeEventValidator;
 
     @Override
     public ListenableFuture<Void> saveAsync(EdgeEvent edgeEvent) {
+        boolean isEdgeEventTenantRateLimitReached = !rateLimitService.checkRateLimit(LimitedApi.EDGE_EVENTS, edgeEvent.getTenantId());
+        boolean isEdgeEventRateLimitPerEdgeReached = !rateLimitService.checkRateLimit(LimitedApi.EDGE_EVENTS_PER_EDGE, edgeEvent.getTenantId(), edgeEvent.getEdgeId());
+        if (isEdgeEventTenantRateLimitReached || isEdgeEventRateLimitPerEdgeReached) {
+            throw new TbRateLimitsException(EntityType.EDGE);
```

Comment:
EntityType.EDGE is per-edge rate limits are exceeded, EntityType.TENANT otherwise